src/utils/rand/mod.rs

```diff
@@ -0,0 +1,65 @@
+// Copyright 2019-2025 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use rand::{CryptoRng, Rng, RngCore, SeedableRng as _};
+
+/// A wrapper of [`rand::thread_rng`] that can be overridden by reproducible seeded
+/// [`rand_chacha::ChaChaRng`] via `FOREST_TEST_RNG_FIXED_SEED` environment variable
```

Comment:
It's helpful to add reasoning on `why` this is used/needed so that future archaeologists can understand it. 
```suggestion
/// A wrapper of [`rand::thread_rng`] that can be overridden by reproducible seeded
/// [`rand_chacha::ChaChaRng`] via `FOREST_TEST_RNG_FIXED_SEED` environment variable. This is required for reproducible test cases for normally non-deterministic methods.
```